clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2831,6 +2830,10 @@ scheduler_tmpl_t<RecordType, ReaderType>::set_cur_input(output_ordinal_t output,
                     return status;
             }
         }
+        if (options_.mapping == MAP_TO_ANY_OUTPUT && prev_input != input &&
```

Comment:
The order of the last_run_time update vs the add_to_ready_queue() call is apparently important, so can you add a comment explaining why?  Otherwise, it seems possible some future editor might unwittingly reorder these code blocks.

Reply:
I think it's still safe the other way, but this just feels better.  I wrote:

            // Now that we've updated prev_info, add it to the ready queue (once on the
            // queues others can see it and pop it off, though they can't access/modify
            // its fields (for identifying if it can be migrated, e.g.) until we release
            // the input lock, so it should be safe to add first, but this order is more
            // straightforward).

Also moved it inside the same if() and removed the redundant check prev_input!=input to make the coupling even clearer.